apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java

```diff
@@ -435,4 +438,117 @@ public Mono<BiInformationResponse> getExperimentBIInformation() {
+                                            .promptId(promptId)
+                                            .name(name)
+                                            .build())
+                                    .flatMap(experimentPage -> enrichExperiments(experimentPage.content())
+                                            .map(experiments -> {
+                                                var groupedResponse = new Experiment.GroupedExperimentResponse(dataset,
+                                                        experiments, experimentPage.total());
+                                                return Experiment.GroupedExperimentPage.builder()
+                                                        .page(page)
+                                                        .size(size)
+                                                        .total(1) // Only one dataset group
+                                                        .content(List.of(groupedResponse))
+                                                        .sortableBy(sortingFactory.getSortableFields())
+                                                        .build();
+                                            }));
+                        });
+            });
+        }
+
+        // Fetch datasets with experiments only
+        return Mono.deferContextual(ctx -> {
+
+            // Build dataset criteria
+            var datasetCriteria = com.comet.opik.api.DatasetCriteria.builder()
+                    .withExperimentsOnly(true)
+                    .promptId(promptId)
+                    .build();
+
+            // Use user-provided sorting for dataset grouping when sorting by dataset fields,
+            // otherwise use default dataset name DESC for consistent grouping
+            var datasetSortingFields = getDatasetSortingFields(sortingFields);
+
+            // Fetch paginated datasets with fixed sorting
+            return Mono
+                    .fromCallable(() -> datasetService.find(page, size, datasetCriteria, datasetSortingFields))
+                    .flatMap(datasetPageObj -> {
+                        var datasets = datasetPageObj.content();
+                        if (datasets.isEmpty()) {
+                            return Mono.just(Experiment.GroupedExperimentPage.empty(page,
+                                    sortingFactory.getSortableFields()));
+                        }
+                        // For each dataset, fetch experiments (first page, default size)
+                        // Use user-provided sorting only for experiments within each group
+                        // Use concatMap to preserve dataset order (deterministic sorting)
+                        return Flux.fromIterable(datasets)
+                                .concatMap(dataset -> experimentDAO
+                                        .find(1, 10, com.comet.opik.api.ExperimentSearchCriteria.builder()
```

Comment:
One concern here is scalability. Since we're currently using `concatMap`, each dataset is processed strictly sequentially, meaning the next call won't start until the previous one completes. This guarantees ordering but can significantly impact latency, especially if experimentDAO.find(...) or the enrichment are slow.

We could consider using `flatMapSequential`, that would allow concurrent execution while preserving the order of results in the final list.